lib/devise/models/authenticatable.rb

```diff
@@ -253,6 +253,7 @@ def find_or_initialize_with_error_by(attribute, value, error=:invalid) #:nodoc:
 
         # Find or initialize a record with group of attributes based on a list of required attributes.
         def find_or_initialize_with_errors(required_attributes, attributes, error=:invalid) #:nodoc:
+          attributes = attributes.to_unsafe_h.with_indifferent_access if attributes.respond_to? :to_unsafe_h
```

Comment:
What I've found is that both `DeviseController#resource_params` and the `ParameterSanitizer` are passing down instances of `ActionController::Parameters` to the model methods, and we operate on them like if they were regular HWIA - which they aren't anymore. 

I think it might be worth some refactor on the sanitizer class, but I want to deal with that after the PR, and I'm not sure if we should change the `resource_params` method as discussed back in #3162.